### PR TITLE
Add the contribution guide

### PR DESCRIPTION
**Pull request by NeilKleistGao**
Related Issue: #137 

- Add error message when TypeScript is not installed
- Add Getting Started Guide in `README.md`

### CODE REVIEW

**Comment by NeilKleistGao** [comment]
Tell me if more details need to be added. 😺

File: README.md [hunk 1]
```diff
@@ -7,14 +7,78 @@ These features can be used to implement expressive class hierarchies as well as
...
+- The `ts2mls/js/src/test/typescript` directory contains the TypeScript test code.
+
+- The `ts2mls/js/src/test/diff` directory contains the declarations generated by ts2mls.
+
+### Prerequisites
+
+To compile the project and run the tests, all you need is a recent Java Virtual Machine (JVM), the Scala Build Tool (SBT), NodeJS, and TypeScript.
+
+The simplest way to set up the JVM and SBT is through the `coursier` tool.
+We explicitly support NodeJS versions `v16.14.x`, `v16.15.x`, `v16.16.x`, and `v17.x.x`.
```
Comment (chengluyu) [inline README.md]: We also tested `v18` if I remembered correctly?

**Comment by LPTK** [comment]
Cool! You can also mention that individual tests can be run with `-z`. For example, `~mlscriptJVM/testOnly mlscript.DiffTests -- -z parser` will watch for file changes and continuously run all parser tests (those that have `"parser"` in their name).

File: ts2mls/js/src/main/scala/ts2mls/TSCompilerInfo.scala [hunk 2]
```diff
@@ -7,7 +7,13 @@ import js.JSConverters._
...
 
 object TypeScript {
-  private val ts = g.require("typescript")
+  private val ts: js.Dynamic = try g.require("typescript") catch {
+    case _ : Throwable => {
+      System.err.println("Cannot find typescript in the current directory. Please install by running \"npm install\".")
+      val process = g.require("process")
+      process.exit(-1)
+    }
+  }
```
Comment (chengluyu) [inline ts2mls/js/src/main/scala/ts2mls/TSCompilerInfo.scala:16]: Is this change related to this PR? Or did I miss something?

File: README.md [hunk 3]
```diff
@@ -7,14 +7,78 @@ These features can be used to implement expressive class hierarchies as well as
...
+```scala
+private val testsData = List(
+    // Put all input files in the `Seq`
+    // Then indicate the output file's name
+    (Seq("Array.ts"), "Array.d.mls")
+  )
+```
 
-## Running the demo locally
+### Running the demo locally
```
Comment (chengluyu) [inline README.md]:
```suggestion
### Running the web demo locally
```

File: README.md [hunk 4]
```diff
@@ -7,14 +7,78 @@ These features can be used to implement expressive class hierarchies as well as
...
+You can also run all tests simultaneously.
+In the terminal, run `sbt test`.
+
+### Running tests individually
+
+To run the tests in main MLScript project individually,
+you can indicate the test you want in `shared/src/test/scala/mlscript/DiffTests.scala`:
+
+```scala
+// Allow overriding which specific tests to run, sometimes easier for development:
```
Comment (chengluyu) [inline README.md]:
```suggestion
  // Allow overriding which specific tests to run, sometimes easier for development:
```

You forgot the leading spaces. 😛

**Review by chengluyu — CHANGES_REQUESTED** [review]

File: ts2mls/js/src/main/scala/ts2mls/TSCompilerInfo.scala (hunk 2 above)
Comment (NeilKleistGao) [reply ts2mls/js/src/main/scala/ts2mls/TSCompilerInfo.scala:16]: Not strongly related. But as required I added some error information. Maybe I should change the title of this PR, or simply use another PR to submit this change.

Comment (chengluyu) [reply ts2mls/js/src/main/scala/ts2mls/TSCompilerInfo.scala:16]: I think it's fine.

File: README.md (hunk 1 above)
Comment (NeilKleistGao) [reply README.md]: I copied this information from [getting-started.md](https://github.com/hkust-taco/mlstruct/blob/mlstruct/getting-started.md). My node's version is also `17.x.x` so I'm not sure. :joy:

Comment (LPTK) [reply README.md]: Just look at NodeTest.scala. I think it now supports it.

**Review by LPTK — APPROVED** [review]
LGTM, thanks! But before you merge (by squashing and cleaning up the commit message), please describe the `-z` way as the main way to run individual tests (put it first). The other way is more of an ugly hack.

File: README.md (hunk 1 above)
Comment (NeilKleistGao) [reply README.md]: Yes indeed.

**Comment by NeilKleistGao** [comment]
Seems I can't merge before Luyu review again. 🤔

**Comment by chengluyu** [comment]
Wait a second. I have some changes.

**Review by chengluyu — APPROVED** [review]

**Pull request merged**